insights/tests/test_json_parser.py

```diff
@@ -8,11 +8,16 @@ class MyJsonParser(JSONParser):
     pass
 
 
-def test_json_parser_success():
+def test_json_parser_object_success():
     ctx = context_wrap('{"a": "2"}')
     assert MyJsonParser(ctx)
 
 
+def test_json_parser_array_success():
+    ctx = context_wrap('["foo", {"bar":["baz", null, 1.0, 2]}]')
+    assert MyJsonParser(ctx)
+
+
 def test_json_parser_failure():
     ctx = context_wrap("boom")
     with pytest.raises(ParseException) as ex:
```

Comment:
@fcami would you mind to add some additional tests here to make sure not only that the parser returns an object, but that the object contains the correct values.  Those tests should have been in the origin parser, but somehow slipped by.  Also please add tests to confirm that the `LegacyItemAccess` mixin, which assumes that `self.data` is a dictionary will work with a `list` result.

Reply:
hi @bfahr I've pushed tests designed the way you wanted for JSONParser.
However wrt LegacyItemAccess I really don't have the knowledge of how it should work.